Fix libva requirements for rocdecode

Mystro256 opened the pull request:
mesa-amdgpu-va-drivers is built with libva 2.16 (VA-API 1.16), so it provides the entry point "__vaDriverInit_1_16". For rocdecode to use mesa, it also needs to make sure it has a high enough requirement on libva to be compatible with this function.

Strictly speaking, it doesn't matter what libva is used as long as it's 2.16 or newer, since libva is backwards compatible. An OR conditions is used to favour distro packages when possible to avoid causing issues with existing libraries built against the distro version.

For libva dev packages, we can just use libva-amdgpu-dev/el directly.

Mystro256 committed:
Fix libva requirements for rocdecode

mesa-amdgpu-va-drivers is built with libva 2.16 (VA-API 1.16), so it
provides the entry point "__vaDriverInit_1_16". For rocdecode to use
mesa, it also needs to make sure it has a high enough requirement on
libva to be compatible with this function.

Strictly speaking, it doesn't matter what libva is used as long as it's
2.16 or newer, since libva is backwards compatible. An OR conditions is
used to favour distro packages when possible to avoid causing issues
with existing libraries built against the distro version.

For libva dev packages, we can just use libva-amdgpu-dev/el directly.

Signed-off-by: Jeremy Newton <Jeremy.Newton@amd.com>

AryanSalmanpour reviewed (changes requested):
@Mystro256, please also make the following changes into [rocDecode-setup.py](https://github.com/ROCm/rocDecode/blob/develop/rocDecode-setup.py). DevOP uses this file to install the prerequisite for rocDecode when building it from the source.

1. Change [libva-dev](https://github.com/ROCm/rocDecode/blob/develop/rocDecode-setup.py#L176) to `libva-amdgpu-dev` in line 176 for `coreDebianPackages`
2. Add `libva2-amdgpu, libva-amdgpu-drm2, libva-amdgpu-wayland2, libva-amdgpu-x11-2` after [line 182](https://github.com/ROCm/rocDecode/blob/develop/rocDecode-setup.py#L182) for `runtimeDebianPackages`
3. Change [libva-devel](https://github.com/ROCm/rocDecode/blob/develop/rocDecode-setup.py#L195) to `libva-amdgpu-devel` in line 195 for `coreRPMPackages` 
4. Add libva-amdgpu after [line 202](https://github.com/ROCm/rocDecode/blob/develop/rocDecode-setup.py#L202) for `runtimeRPMPackages`

Changes in main [README.md](https://github.com/ROCm/rocDecode/blob/develop/README.md)

1. Change this section (Video Acceleration API Version 1.5.0 or later - Libva is an implementation for VA-API) to Video Acceleration API Version 2.16.0 or later - Libva is an implementation for VA-API 
2. Change` sudo apt install libva-dev` to `sudo apt install libva-amdgpu-dev`
3. Change `sudo apt install mesa-amdgpu-va-drivers` to `sudo apt install libva2-amdgpu libva-amdgpu-drm2 libva-amdgpu-wayland2 libva-amdgpu-x11-2 mesa-amdgpu-va-drivers`


Changes in [CHANGELOG.md](https://github.com/ROCm/rocDecode/blob/develop/CHANGELOG.md)

1.  Add a bullet in CHANGELOG.md describing the changes made in this PR (e.g., VA driver dependencies updated)

Mystro256 committed:
Update to use libva-amdgpu

To reflect the package change, update the README, rocDecode-setup.py,
and the CHANEGLOG.

Putting the minimum VA-API version in the README isn't required as the
user is expected to just install the latest libva-amdgpu to match the
mesa VA-API version.

Mystro256 commented:
> @Mystro256, please also make the following changes...

Ok done, I pushed a second commit, please let me know if I missed anything.

samjwu commented:
confirmed pass on sles http://math-ci.amd.com/job/mainline/job/precheckin/job/rocDecode/job/PR-435/5/pipeline-graph/

rrawther commented on CHANGELOG.md:
```diff
@@ -21,8 +21,7 @@ Documentation for rocDecode is available at
 * Package - VA driver dependencies updated
-  * VA drivers - only use mesa-amdgpu-va-drivers for both debian and RPM based OS
-  * mesa-amdgpu-va-drivers - brings all it's dependencies
+  * VA drivers - libva-amdgpu packages are now used to avoid API related issues
```
Probably need to mention what is removed and what is added as new so it is clearer

rrawther reviewed (approved):
Added a comment to modify the changelog to be more clear
